### PR TITLE
Add user workspace validations

### PR DESCRIPTION
**Pull request by frascuchon**
This PR replicates workspace/user validations using new DB models.

This PR shows where to define the validations, and how to use them in handlers. Once this is well defined/validated, we will  remove the old validation implemented in pydantic `User` class (`check_workspace`...) in a separate PR.

### CODE REVIEW

File: src/argilla/server/contexts/validations.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        if workspace in user.workspaces:
+            continue
+        elif accounts.is_superuser(user):
+            continue
+        else:
+            raise ForbiddenOperationError(f"Cannot access workspace {workspace}")
+    return True
+
+
+def validate_and_get_workspace_by_name(db: SessionLocal, workspace_name: str) -> Workspace:
```
Comment (jfcalvo) [inline src/argilla/server/contexts/validations.py]: @frascuchon we can create a function on `accounts` context with the following code:

```python
def get_workspace_by_name_or_raise(db: Session, workspace_name: str):
    workspace = get_workspace_by_name(db, workspace_name)

    if not workspace:
        raise EntityNotFoundError(name=str(workspace.id), type=Workspace)

    return workspace
```

Comment (frascuchon) [reply src/argilla/server/contexts/validations.py]: Done for workspace_by_name. See the code [here](https://github.com/argilla-io/argilla/blob/a5b3294a3423d0724ab88749335669e252d35393/src/argilla/server/contexts/accounts.py#L63)

File: src/argilla/server/contexts/validations.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from argilla.server.contexts import accounts
+from argilla.server.models import User, Workspace
+
+
+def can_user_access_to_workspace(user: User, workspace: Workspace):
```
Comment (jfcalvo) [inline src/argilla/server/contexts/validations.py]: I think we need more granularity with the permissions. Maybe something like `is_authorized(user, "create", workspace)` or something similar, "access" is a little bit ambiguous.

In any case we can have a discussion about this.

Comment (frascuchon) [reply src/argilla/server/contexts/validations.py]: I have done some changes here. We have no functional pattern matching in python so, this kind of function definition could grow easily when we need to authorize several resources for a user. Anyway, let's discuss this from here on Monday.

**Comment by jfcalvo** [comment]
Closing. A different approach has been defined on https://github.com/argilla-io/argilla/pull/2491

**Pull request closed without merging**